pkg/container/bitlpm/trie.go

```diff
@@ -101,16 +101,6 @@ type Trie[K Key[K], T any] interface {
 	ForEach(fn func(uint, K, T) bool)
 }
 
-// Iterator is an interface that can be used to produce the next key/value pair in iteration
```

Comment:
I generally don't prefer  returning unexported types, because it can lead to awkwardness where variable types cannot be spelled. Thus, one must always use `:=`.

```go
i := trie.AncestorIterator()
```

For example, this code becomes impossible:
```go
var i trie.Iterator 
if foo {
  i = trie.AncestorIterator()
} else {
  i = trie.AncestorLongestPrefixFirstIterator()
}
...

However, the stack allocation is useful. Maybe keep the Iterator interface for convenience?

Reply:
And maybe we should add a TODO to turn these in to rangefuncs?

Reply:
Nested range funcs have horrible performance (https://github.com/golang/go/issues/69015). These iterators are used from rangefuncs and this combination has the best performance and the business logic using the (outer) rangefunc iterators is very concise.